### ISSUE TITLE
Feature: anticheat

### ISSUE THREAD

**Issue by FrancescoBorzi** [azerothcore/azerothcore-wotlk#4007]
There is a [PR from kvpika](https://github.com/azerothcore/azerothcore-wotlk/pull/2088) that has never been completed and merged. It would be nice to complete it by adapting it to the latest AC.

It would be nice to implement these new features to the mod-anti-cheat that we already have here: 

https://github.com/azerothcore/mod-anticheat

It would be great to use that one and create configurations to enable/disable certain checks. If new hooks are needed in order to accomplish this, they can certainly be added to the core.

**Comment by IceNet** [comment]
There's alrdy a passive anti-cheat module.

**Comment by Helias** [comment]
Yes, but it does not the same checks of that PR, I think we need the kvipka's PR in the core

**Comment by Viste** [comment]
> Yes, but it does not the same checks of that PR, I think we need the kvipka's PR in the core

for that you need remove position hardcodes from almost all scripts and make them dynamic like in TC and mangos

**Comment by Winfidonarleyan** [comment]
> There's alrdy a passive anti-cheat module.

Totally useless module

**Comment by disclosurez** [comment]
I use kvipka's anticheat over at wotlkraids, I can provide feedback to this anticheat and fixes to false positives where I have found them

**Comment by Ramys** [comment]
it would be interesting for you to post here.

**Comment by disclosurez** [comment]
> 
> 
> it would be interesting for you to post here.

if someone wants to merge the existing PR with the current build then I can comment on my findings.

**Comment by Yehonal** [comment]
please guys, to implement these new features, reuse the mod-anticheat that we already have here: https://github.com/azerothcore/mod-anticheat 

You can open a PR there changing the readme etc. but would be great to use that one and creating conf to enable/disable certain checks...instead of creating a new anticheat from scratch.

**Comment by FrancescoBorzi** [comment]
> please guys, to implement these new features, reuse the mod-anticheat that we already have here: https://github.com/azerothcore/mod-anticheat
> 
> You can open a PR there changing the readme etc. but would be great to use that one and creating conf to enable/disable certain checks...instead of creating a new anticheat from scratch.

@Yehonal I agree with it and I've updated my original post.

**Comment by FrancescoBorzi** [comment]
@UltraNix would you like to work on this? preferably porting this into a module, such as: https://github.com/azerothcore/mod-anticheat

**Comment by UltraNix** [comment]
Working on ObjectGuid port - it will take some time.

**Comment by UltraNix** [comment]
@FrancescoBorzi I will try to port that - assign me.

**Comment by FrancescoBorzi** [comment]
thanks @UltraNix , done

**Comment by FrancescoBorzi** [comment]
is anyone actively working on this?

**Comment by Nyeriah** [comment]
> is anyone actively working on this?

@acidmanifesto

**Comment by acidmanifesto** [comment]
I am currently maintaining the current anticheat module as of recently and as of late  (Not the KvipKA version) based on feed back from the GM's who have to rely on it.
Althought KvpiKA is unique and the pr was useful at its time, without propper maintaining of the anticheat it has show it nothing but to be filled with false positives or no detectections and has sense became stagnant, out of date, and unreliable, which in other words makes it irrelevant. Not this is only based on that one closed PR that was made by Kvpika https://github.com/azerothcore/azerothcore-wotlk/pull/2088 and the attempted refactor into a module here https://github.com/azerothcore/mod-anticheat/pull/32 .
Now currently the Anticheat we are using, with massive feedback from @55Honey  and crew. False detection has gone down, Cheaters Propperly Identified, and New and unique detections has been incorporated and exsisting detections being updated based on our source.

**Major Difference between Kvpika and the Current Anticheat Module and some short facts:**
**Kvipka:**
Kvpika anticheat, was a paid module and was made free based on resellers who leaked , it was maded available as a PR from Kvipka on tc and azerothcore, I was to understand the current anticheat maintained by Kvpika is a paid one and is based on TC support with refactoring to the core for Azerothcore.

**It offers:**
SpeedHack: more hits with false positives in many MANY scenarios, including riding transports and falling
FlyHack Detected: Reliable, but false hits on transports and falling.
DoubleJump: Reliable due to it comparing opcode for jump
Fake Jumper: more hits with false positives in many MANY scenarios, such as falling and riding transports at time.
FakeFlying: Same as above and uncertain of it's difference
Wallclimb: detects about 30% of the time when using cheat, loosely reliable
Ignore control: Reliable
NoFallingDamage: Currently impossible to evaluated on Azerothcore due to the cores handling, flight hack will have to be used
WaterWalk: Currently Does not detect at all on current azerothcore, relied on a depreciated water walking opcode which was removed in 3.0.1

**Logging** is a mess and offers auto penalties which will get issues out alot based on false positives that is the common complaint with this module. It does offer logging in world console and as a seperate log.

**Auto Applied Penalty:
It has a conf to auto apply a Penalty such as kick and banned based on auto matic reporting, Can kick and ban users based on false postivie hits alone.

**Azerothcore Current Anticheat module:**
Origins no fully cleared but is known and was known as Passive Anticehat from TC and was first established over 14years ago and has been the base of alll things anticheat.

Since my recent involvement on anticheat in order to help out Chromie Craft staff.
We got it working with detection again. Highly Reduce False Positives, New cheat detections introduced. Improvements based on current revision of Azerothcore and has relevance.

**Azerothcore Anticheat:**
Ours offers the current detections:

FlyHack: Reliable, better detection with stricter fly hack detection enabled in conf
WaterWalkHack: We removed a few additional false hits, still needing to work on better detections but does detect when a exploit is used. Can only work based on a certain spell exploitation as the opcode usuage will not work on 3.3.4 as the opcode was depreciated and unused in 3.0.1.
JumpHack: Reliable, Compares previous and current jump opcode being used
TelePlaneHack: Reliable, Although false posivites occur when falling from great distance.
SpeedHack: Reliable and adjusted formular to exclude false hits on falling
ClimbHack: Adjust formula where it now detects 80% of the hack being active.
TelePortHack: Highly Reliable with the false hit being landing at flight managers.
Ignore ControlHack: New introductions to the anticheat, we are still working on the details of what is consider a false positive with this one.

**Logging:**
Raid message spam on screen for GM's Logging and currently GM Chat message is sent to any available gms in game with a hyper link name of the player in the game chat for the gm to click on. Server.log is also logged as well anything on the world console.

**Auto Applied Penalty:**
We offer only two auto applied penalties and that is when enabled in the conf. That is to kick the player for water walk hack and fly hack detection.

**My Opinions on this Matter:
TLDR:**
Cancel the Bounty and Put it towards any other PR that requires attention or Fund Arctium for their assistance with sniffer\parser support and their contributions to the wow scene. We can maintain our current anticheat module just fine since i had some time and live feed back from CC staff.

**Long Version:**
Lets stick to our current open source, we are maintaining it better and introduced new detections and changes. We are able to maintain the current exsisting code easily. Kvipka, even though i am certain the pr at the time was great in comparison with the anticheat cheat that was current at the time, it was not our original code, there could be ethics debates based on it because his current version is a paid version and use working off a free version may be looked at in the wrong light, too many false hits, out of date detection, not wow emulations original open source code, so issues may come about in the future for maintaining it, like pretty much how the current one went stagnant until i got involved into it a couple of weeks ago and got it reworked and improved upon. Any exsisting bounty the funding should be reinvested in other areas of azerothcore or the wow scene with key contributors.
